File: 1441.build-an-array-with-stack-operations.py
#
# @lc app=leetcode id=1441 lang=python3
#
# [1441] Build an Array With Stack Operations
#

# @lc code=start
from typing import List
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        #Stack s =    bottom ->[1,2,3,4,5] <- top
        ans = []
        m = len(target)
        i, j = 1, 0
        while i <= n and j < m:
            ans.append("Push")
            if target[j] != i:
                ans.append("Pop")
            else:
                j += 1
            i += 1
        return ans

# @lc code=end

File: 1441.build-an-array-with-stack-operations.java
/*
 * @lc app=leetcode id=1441 lang=java
 *
 * [1441] Build an Array With Stack Operations
 */

// @lc code=start
import java.util.ArrayList;
import java.util.List;
class Solution {
    public List<String> buildArray(int[] target, int n) {
        //clarification 
        //we need to fill and modify an empty stack with n integers to make the stack == target, return the operations (Push Pop) needed

        //constriants
        //1 <= target.length <= 100
        //target is increasing
        //1 <= n <= 100
        //1 <= target[i] <= n

        List<String> ans = new ArrayList<>();
        int mx = target[target.length - 1];
        int i = 0;
        for(int x = 1; x <= mx; x++){
            ans.add("Push");
            if(x == target[i]){
                i++;
            }
            else{
                ans.add("Pop" );
            }
        }
        return ans;
    }
}
// @lc code=end

